src/components/CountryPicker/CountrySelectorModal.js

```diff
@@ -0,0 +1,93 @@
+import _ from 'underscore';
+import React, {useState, useMemo} from 'react';
+import PropTypes from 'prop-types';
+import * as PersonalDetails from '../../libs/actions/PersonalDetails';
+import CONST from '../../CONST';
+import useLocalize from '../../hooks/useLocalize';
+import HeaderWithBackButton from '../HeaderWithBackButton';
+import SelectionListRadio from '../SelectionListRadio';
+import Modal from '../Modal';
+
+const propTypes = {
+    /** Whether the modal is visible */
+    isVisible: PropTypes.bool.isRequired,
+
+    /** Country value selected  */
+    currentCountry: PropTypes.string.isRequired,
+
+    /** Function to call when the user selects a Country */
+    onCountrySelected: PropTypes.func,
+
+    /** Function to call when the user closes the Country modal */
+    onClose: PropTypes.func,
+};
+
+const defaultProps = {
+    onClose: () => {},
+    onCountrySelected: () => {},
+};
+
+function filterOptions(searchValue, data) {
+    const trimmedSearchValue = searchValue.trim();
+    if (trimmedSearchValue.length === 0) {
+        return [];
+    }
+
+    return _.filter(data, (country) => country.text.toLowerCase().includes(searchValue.toLowerCase()));
+}
+
+function CountrySelectorModal({currentCountry, isVisible, onClose, onCountrySelected}) {
+    const {translate} = useLocalize();
+    const selectedSearchCountry = PersonalDetails.getCountryName(currentCountry);
+    const [searchValue, setSearchValue] = useState(selectedSearchCountry);
+
+    const countries = useMemo(
+        () =>
+            _.map(translate('allCountries'), (countryName, countryISO) => ({
+                value: countryISO,
+                keyForList: countryISO,
+                text: countryName,
+                isSelected: currentCountry === countryISO,
+            })),
+        [translate, currentCountry],
+    );
+
+    const filteredData = filterOptions(searchValue, countries);
+    const headerMessage = searchValue.trim() && !filteredData.length ? translate('common.noResultsFound') : '';
```

Comment:
I prefer just to memo the countries so I avoid the map creation on every key press, that stay stable.

On the other hand, memorizing the filter won't add any benefits because `searchValue` changes after every key press. So the values will be recreated every time.

Reply:
Good point

Reply:
This can be marked as resolved.